File: fcc/js/aug/fibonacci.js
function fibonacciSequence(startSequence, length) {
  if (length < 1) {
    return [];
  }

  let answer = [];

  if (length < 3) {
    for (let c = 0; c < length; c++) {
      answer.push(startSequence[c]);
    }
    return answer;
  }
  let before = startSequence[0];
  let current = startSequence[0];
  let next = startSequence[1];

  answer.push(before);
  answer.push(next);

  for (let i = 0; i < length - 2; i++) {
    before = current;
    current = next;
    next = before + next;
    answer.push(next);
  }
  length = answer;
  return length;
}
window.alert(fibonacciSequence([123456789, 987654321], 5));

File: fcc/cpp/aug/fibonacci.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<long long int> fibonacciSequence(int startSequence[2], int length)
{
	if (length < 1)
		return {};

	vector<long long int> answer;
	if (length < 3)
	{
		for (int c = 0; c < length; c++)
		{
			answer.push_back(startSequence[c]);
		}
		return answer;
	}
	long long int before = startSequence[0];
	long long int current = startSequence[0];
	long long int next = startSequence[1];
	answer.push_back(current);
	answer.push_back(next);
	for (int i = 0; i < length - 2; i++)
	{
		before = current;
		current = next;
		next = before + current;
		answer.push_back(next);
	}

	return answer;
}